### ISSUE TITLE
Update GraalVM to 21.3.0

### ISSUE THREAD

**Issue by 4e6** [enso-org/enso#3139]
### Summary

There is a new version of [GraalVM](https://github.com/graalvm/graalvm-ce-builds/releases/tag/vm-21.3.0) released, and we need to update our project.

### Value

- New GraalVM brings smaller native images with faster startup time
- Project don't have outdated dependencies

### Specification

- [ ] Run runtime benchmarks before upgrading with `sbt runtime/bench`
- [ ] Bump GraalVM version in `build.sbt`, `Dockerfile`, and `signArchives.js` ide task. (you can grep for `21.1.0`)
- [ ] As a sanity check, run the runtime benchmarks again after the upgrade. The benchmark will end unsuccessfully if the new values are larger than the previous run.

### Acceptance Criteria & Test Cases

- [ ] The build uses the new version of Graal

**Comment by 4e6** [comment]
The following benchmarks are slower on GraalVM 21.3:

```
AtomBenchmarks.benchGenerateListQualified  avgt    5  36.267 ± 6.978  ms/op
Difference was: 2.5553439948976147.
AtomBenchmarks.benchMapReverseCurryList  avgt    5  100.982 ± 264.493  ms/op
Difference was: 5.531108522099395.
AtomBenchmarks.benchMapReverseList  avgt    5  29.247 ± 2.075  ms/op
Difference was: 0.8277674962556598.
AtomBenchmarks.benchReverseList  avgt    5  39.543 ± 9.111  ms/op
Difference was: 2.0437120393553165.
AtomBenchmarks.benchReverseListMethods  avgt    5  29.923 ± 0.251  ms/op
Difference was: 0.9418725077273267.
AtomBenchmarks.benchSumList  avgt    5  34.639 ± 0.146  ms/op
Difference was: 4.3122296860367015.
AtomBenchmarks.benchSumListFallback  avgt    5  20.248 ± 0.034  ms/op
Difference was: 2.139181032707289.
AtomBenchmarks.benchSumListMethods  avgt    5  21.175 ± 0.606  ms/op
Difference was: 2.218395766264448.
AtomBenchmarks.sumListLeftFold  avgt    5  19.664 ± 0.462  ms/op
Difference was: 2.0615759567352763.
NamedDefaultedArgumentBenchmarks.benchSumTCOWithDefaultArgs  avgt    5  3096.510 ± 11.298  ms/op
Difference was: 30.466575897449715.
NamedDefaultedArgumentBenchmarks.benchSumTCOWithNamedArgs  avgt    5  1524.591 ± 2.492  ms/op
Difference was: 12.433576729048243.
RecursionBenchmarks.benchNestedThunkSum  avgt    5  1607.139 ± 3.758  ms/op
Difference was: 9.084785927106383.
RecursionBenchmarks.benchOversaturatedRecursiveCall  avgt    5  1543.062 ± 4.217  ms/op
Difference was: 13.804322350323977.
RecursionBenchmarks.benchSumRecursive  avgt    5  0.002 ±  0.001  ms/op
Difference was: 0.3988810706003003.
RecursionBenchmarks.benchSumStateTCO  avgt    5  1538.804 ± 6.431  ms/op
Difference was: 13.699444022161448.
RecursionBenchmarks.benchSumTCO  avgt    5  1514.640 ± 5.489  ms/op
Difference was: 12.309377660525977.
RecursionBenchmarks.benchSumTCOFoldLike  avgt    5  3004.540 ± 5.884  ms/op
Difference was: 27.76485658359691.
RecursionBenchmarks.benchSumTCOWithEval  avgt    5  1518.179 ± 2.756  ms/op
Difference was: 12.282425943927796.
```

[bench-report.zip](https://github.com/enso-org/enso/files/7596209/bench-report.zip)

**Comment by kustosz** [comment]
1. I believe this upgrade should be performed version-by-version (as we're already 4 versions behind the newest 22.1.0).
2. After each version the benchmarks should be checked. That includes both the java-side benchmarks (`runtime/bench` target) and the in-language benchmarks implemented in https://github.com/enso-org/enso/tree/develop/test/Benchmarks
3. Each slowdown should be either fixed or its reason documented, with a possible path forward to fixing it.
4. Looking at the release notes, there are a number of things that definitely should be updated:
    1. The existing safepoint system (which is fancy speak for "check a volatile boolean") should be replaced by Truffle's new system.
    2. When migrating to 21.3.0, all `Context` accesses should be migrated to the new pattern (i.e. a static method on `Context`).
    3. When migrating to 21.2.0 all nodes should be verified for pointless uses of `@Specialization` instead of `@Fallback`, that were previously in place due to `@Fallback` not supporting `@Cached` and `@CachedLibrary`
    4. When migrating to 22.0.0, frame access patterns will need to be updated.
    5. In general, any deprecations should be immediately fixed.

**Comment by 4e6** [comment]
Implemented in #3258
